File: app/src/main/java/com/bungabear/inubus/Fragment/DestinationFragment.java
package com.bungabear.inubus.Fragment;

import android.content.Context;
import android.os.Bundle;
import android.support.annotation.Nullable;
import android.support.v4.app.Fragment;
import android.support.v7.widget.LinearLayoutManager;
import android.support.v7.widget.RecyclerView;
import android.view.LayoutInflater;
import android.view.View;
import android.view.ViewGroup;

import com.bungabear.inubus.Custom.DestinationRecyclerAdapter;
import com.bungabear.inubus.R;

/**
 * Created by Bunga on 2018-01-29.
 */

public class DestinationFragment extends Fragment {

    private Context context;
    private RecyclerView rv;
    public static DestinationFragment newInstance(Context context) {
        Bundle args = new Bundle();
        DestinationFragment fragment = new DestinationFragment();
        fragment.setArguments(args);
        fragment.context = context;
        return fragment;
    }

    @Override
    public void onCreate(@Nullable Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
    }

    @Nullable
    @Override
    public View onCreateView(LayoutInflater inflater, @Nullable ViewGroup container, @Nullable Bundle savedInstanceState) {
        View v = inflater.inflate(R.layout.fragment_destination, container, false);
        rv = v.findViewById(R.id.destination_recycler_list);

        LinearLayoutManager mLayoutManager = new LinearLayoutManager(context);
        rv.setLayoutManager(mLayoutManager);

        DestinationRecyclerAdapter adapter = new DestinationRecyclerAdapter();
        for (int i = 0; i < 30; i++) {
            if (i % 4 == 0) {
                adapter.addSectionHeader(""+i);
            }
            adapter.addItem(new String[]{"8","780","908","780-1"} , i%4 + 1);
        }
        rv.setAdapter(adapter);
        rv.getRecycledViewPool().setMaxRecycledViews(0, 0);
        return v;
    }
}

File: app/src/main/java/com/bungabear/inubus/Fragment/DestinationFragment.kt
package com.bungabear.inubus.fragment

import android.content.Context
import android.os.Bundle
import android.support.v4.app.Fragment
import android.support.v4.app.FragmentManager
import android.view.LayoutInflater
import android.view.View
import android.view.ViewGroup
import com.bungabear.inubus.R

/**
 * Created by Minjae Son on 2018-08-13.
 */

class DestinationFragment : Fragment() {

    private lateinit var mFm: FragmentManager
    private lateinit var mContext : Context

    companion object {
        fun newInstance(fm : FragmentManager, context : Context) : DestinationFragment{
            val fragment = DestinationFragment()
            fragment.mFm = fm
            fragment.mContext = context
            return fragment
        }
    }

    override fun onCreateView(inflater: LayoutInflater, container: ViewGroup?, savedInstanceState: Bundle?): View? {
        val view = inflater.inflate(R.layout.fragment_destination, container, false)
        return view
    }
}